src/api/app/models/event_subscription/for_role_form.rb

```diff
@@ -0,0 +1,65 @@
+class EventSubscription
+  class ForRoleForm
+    attr_reader :name, :channels, :subscriber
+
+    def initialize(role_name, event, subscriber)
+      @subscriber = subscriber
+      @name = role_name
+      @event = event
+      @channels = []
+      @subscriptions = []
+    end
+
+    def call
+      @channels = EventSubscription.channels.keys[1..-1].map do |channel|
+        find_subscription_for_event_class_and_role(@event, name, channel)
+      end
+
+      self
+    end
+
+    private
+
+    def find_subscription_for_event_class_and_role(event_class, role, channel)
+      subscriber_subscription =
+        find_subscription_for_subscriber(event_class, role, channel) || find_subscription_for_subscriber(event_class, :all, channel)
+      default_subscription = find_default_subscription(event_class, role, channel)
+
+      # 1. Pick the subscriber's subscription if it exists
+      subscription = if subscriber.present? && subscriber_subscription.present?
+                       subscriber_subscription
+
+                     # 2. Pick the subscriber's subscription if it exists
+                     elsif default_subscription.present?
+                       default_subscription
+
+                     # 3. Otherwise instantiate a new subscription
+                     else
+                       EventSubscription.new(
+                         subscriber: subscriber,
+                         eventtype: event_class.to_s,
+                         receiver_role: role,
+                         channel: 'disabled'
+                       )
+                     end
```

Comment:
I don't think those comments add any additional information while making the formatting hard to parse. Do we really need them?

Reply:
I removed the comments